### PR TITLE
[WIP] rsync on volume

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :gift: Added function that allows to run rsync daemon on volume in cluster and expose it locally using `net.Listener`.
  This can be useful for more efficient source push for on-cluster-build.

/kind enhancement

### CODE REVIEW

**Comment by matejvasek** [comment]
Interesting stuff is in https://github.com/knative/func/pull/1385/commits/3ce37542b4ec6a0fe9b90a96f1390fedb9185f6e

**Comment by matejvasek** [comment]
Still very much WIP.

**Comment by matejvasek** [comment]
@zroubalik you may find this interesting.

**Comment by zroubalik** [comment]
@matejvasek this is cool. My initial idea was to run this from within a Pipeline task, not a temporary pod, so the whole workflow is in sync with the rest of the pipeline. WDYT?

**Comment by matejvasek** [comment]
@zroubalik 
I prefer temporary pod so it is under my control.
Creating it from pipeline would be more complicated.

File: k8s/persistent_volumes.go
Comment (lkingland) [inline k8s/persistent_volumes.go:295]:
```suggestion
			return fmt.Errorf("error while exec (err out: %q): %w", errOut.String(), err)
```

File: k8s/persistent_volumes.go
Comment (lkingland) [inline k8s/persistent_volumes.go:307]:
```suggestion
			return fmt.Errorf("cannot accept connection: %w", err)
```

File: k8s/persistent_volumes_test.go
Comment (lkingland) [inline k8s/persistent_volumes_test.go:208]:
```suggestion
		filepath.Join("testdata", "content", "hello.txt"),
```
Since go tool treats the exact name `testdata` as special, this may be best renamed.  From `go help packages`:
> Directory and file names that begin with "." or "_" are ignored
> by the go tool, as are directories named "testdata".
I am pretty sure it's case-sensitive?

**Review by lkingland — APPROVED** [review]
Looks quite useful.  Zbynek makes a good point, but I do like how straightforward this is.
/lgtm (though maybe I should wait until not a draft to approve?)
/hold for two spelling and one directory rename

File: k8s/persistent_volumes_test.go
Comment (matejvasek) [reply k8s/persistent_volumes_test.go:208]: >I am pretty sure it's case-sensitive

@lkingland  I am not going to argue, but how that would work on non-case sensitive OS?

Comment (matejvasek) [reply k8s/persistent_volumes_test.go:208]: like everything is lower by default?

Comment (matejvasek) [reply k8s/persistent_volumes_test.go:208]: from Go source it looks like it is case-sensitive

**Pull request closed without merging**
